globals/matchmaking_utils.go

```diff
@@ -163,13 +163,29 @@ func CreateSessionByMatchmakeSession(matchmakeSession *match_making_types.Matchm
 	session.GameMatchmakeSession.SessionKey = make([]byte, 32)
 	rand.Read(session.GameMatchmakeSession.SessionKey)
 
+	if session.GameMatchmakeSession.MatchmakeParam == nil {
+		session.GameMatchmakeSession.MatchmakeParam = match_making_types.NewMatchmakeParam()
+	}
+
+	if session.GameMatchmakeSession.MatchmakeParam.Parameters == nil {
+		session.GameMatchmakeSession.MatchmakeParam.Parameters = make(map[string]*nex.Variant)
+	}
+
+	session.GameMatchmakeSession.MatchmakeParam.Parameters["@SR"] = nex.NewVariant()
+	session.GameMatchmakeSession.MatchmakeParam.Parameters["@SR"].TypeID = 3
+	session.GameMatchmakeSession.MatchmakeParam.Parameters["@SR"].Bool = true
+
+	session.GameMatchmakeSession.MatchmakeParam.Parameters["@GIR"] = nex.NewVariant()
+	session.GameMatchmakeSession.MatchmakeParam.Parameters["@GIR"].TypeID = 1
+	session.GameMatchmakeSession.MatchmakeParam.Parameters["@GIR"].Int64 = 3
```

Comment:
This is more just a curiosity than anything, but do we know what purpose these serve?
The official NEX server assigns them (and the Tri-Force Heroes server assigns even more than this), but I've yet to find any client using (or requiring) them.

Reply:
I don't think we know the exact purpose, but they are assigned by the server when creating a session on every title I remember so far

Reply:
I know that Modern Warfare II uses `SR` to mean "Skill Rating" which is used to determine which rank you are and get matched with. I wonder if this is something similar? I know MW2 is not a NEX, or even WiiU, game but I wonder if the terminology is the same/similar? `@SR` being `true` meaning that "Skill Rating" is used when match making, and `@GIR` (**G**athering IDK **R**ating?) is whatever rank?

Reply:
> I know that Modern Warfare II uses `SR` to mean "Skill Rating" which is used to determine which rank you are and get matched with. I wonder if this is something similar? I know MW2 is not a NEX, or even WiiU, game but I wonder if the terminology is the same/similar? `@SR` being `true` meaning that "Skill Rating" is used when match making, and `@GIR` (**G**athering IDK **R**ating?) is whatever rank?

I could buy that if the types were the same, but that isn't the case. It could be something similar though